ln-gateway/build.rs

```diff
@@ -0,0 +1,18 @@
+use std::{env, fs::File, process::Command};
+fn main() {
+    if env::var_os("GIT_HASH").is_none() {
+        let git_hash = if File::open("./../../.git/HEAD").is_ok() {
+            let output = Command::new("git")
+                .args(&["rev-parse", "HEAD"])
+                .output()
+                .unwrap();
+            String::from_utf8(output.stdout).unwrap()
+        } else {
+            // we set a fake hash here and make it easily recognizable
+            // by giving it a prefix (0x0123456)
+            String::from("01234569afbe457afa1d2683a099c7af48a523c1")
+        };
+        println!("cargo:rustc-env=GIT_HASH={}", git_hash);
+    }
+    println!("cargo:rerun-if-env-changed=GIT_HASH");
+}
```

Comment:
Can we deduplicate this build script code somehow? E.g. a new helper crate `fedimint-build`?

Reply:
That's a good idea but I'm not sure if this will be possible. IIRC I saw people complaining that there is no `build = ..` in the `Cargo.toml` which would make that easy. I'll look that up

Reply:
See https://doc.rust-lang.org/cargo/reference/build-scripts.html#build-dependencies

Reply:
Ok there is no `[[bin]] build = ..` but `[package] build = ..` so it works I think